tests/helpers/load_state_tests.py

```diff
@@ -325,24 +302,86 @@ def add_blocks_to_chain(
+    with open(test_file, "r") as fp:
+        data = json.load(fp)
+
+        # Some newer test files have patterns like _d0g0v0_
+        # between test_name and network
+        keys_to_search = re.compile(
+            f"{re.escape(test_name)}.*{re.escape(network)}"
+        )
+        found_keys = list(filter(keys_to_search.match, data.keys()))
+
+        if not any(found_keys):
+            raise KeyError
+
+        for _key in found_keys:
+            yield {
+                "test_file": test_file,
+                "test_key": _key,
+                "test_data": data[_key],
+            }
+
+
 def fetch_state_test_files(
     test_dir: str,
-    slow_test_list: Tuple[str, ...],
-    incorrect_tests_list: Tuple[str, ...],
-    load: BaseLoad,
-) -> Generator[Union[str, ParameterSet], None, None]:
-    for _dir in os.listdir(test_dir):
-        test_file_path = os.path.join(test_dir, _dir)
-        for _file in os.listdir(test_file_path):
-            _test_file = os.path.join(_dir, _file)
-            if _test_file in incorrect_tests_list:
-                continue
-            elif _test_file in slow_test_list:
-                yield pytest.param(_test_file, marks=pytest.mark.slow)
-            else:
-                try:
-                    load_json_fixture(test_dir, _test_file, load)
-                    yield _test_file
-                except KeyError:
-                    # file doesn't contain tests for the given fork
-                    pass
+    network: str,
+    custom_file_list: Tuple[str, ...] = (),
+    slow_list: Tuple[str, ...] = (),
+    ignore_list: Tuple[str, ...] = (),
+) -> Generator[Union[Dict, ParameterSet], None, None]:
+
+    all_slow = [re.compile(x) for x in slow_list]
+    all_ignore = [re.compile(x) for x in ignore_list]
+
+    # Get all the files to iterate over
+    # Maybe from the custom file list or entire test_dir
+    files_to_iterate = []
+    if any(custom_file_list):
+        # Get file list from custom list, if one is specified
+        for _file in custom_file_list:
+            files_to_iterate.append(os.path.join(test_dir, _file))
+    else:
+        # If there isnt a custom list, iterate over the test_dir
+        all_jsons = [
+            y
+            for x in os.walk(test_dir)
+            for y in glob(os.path.join(x[0], "*.json"))
+        ]
+
+        for _full_path in all_jsons:
+            if not any(x.search(_full_path) for x in all_ignore):
+                # If a file or folder is marked for ignore,
+                # it can already be dropped at this stage
+                files_to_iterate.append(_full_path)
+
+    # Start yielding individual test cases from the file list
+    for _test_file in files_to_iterate:
+        try:
+            for _test_case in load_json_fixture(_test_file, network):
+                # identifier could identifiy files, folders or individual cases
+                _identifier = _test_case["test_file"] + _test_case["test_key"]
```

Comment:
Hm, concatenating these together seems like you might get some weird effects?

Reply:
Since the file name always ends with `.json`, I do not really expect any side-effects. 
But you are right. This is not the smartest way to handle this. I have updated the code to handle this using the regex `OR`

`_identifier = "(" + _test_case["test_file"] + "|" + _test_case["test_key"] + ")"`